zio-http-benchmarks/src/main/scala/zhttp.benchmarks/CookieDecodeBenchmark.scala

```diff
@@ -0,0 +1,38 @@
+package zhttp.benchmarks
+
+import org.openjdk.jmh.annotations._
+import zhttp.http._
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.Throughput))
+@OutputTimeUnit(TimeUnit.SECONDS)
+class CookieDecodeBenchmark {
+  val random = new scala.util.Random()
+  val name   = random.alphanumeric.take(100).mkString("")
+  val value  = random.alphanumeric.take(100).mkString("")
+  val domain = random.alphanumeric.take(100).mkString("")
+  val path   = Path((0 to 10).map { _ => random.alphanumeric.take(10).mkString("") }.mkString(""))
+  val maxAge = random.nextLong()
+
+  private val cookie    = Cookie(
+    name,
+    value,
+    Some(Instant.now()),
+    Some(domain),
+    Some(path),
+    true,
+    true,
+    Some(maxAge),
+    Some(Cookie.SameSite.Strict),
+  )
+  private val cookieStr = cookie.encode
+
+  @Benchmark
+  def benchmarkApp(): Unit = {
+    val _ = Cookie.decodeResponseCookie(cookieStr)
```

Comment:
```suggestion
    val _ = Cookie.unsafeDecodeResponseCookie(cookieStr)
```

Reply:
Can you run this one last time and post the results :)

Reply:
Ok, applied your suggestionts.
I have run the benchmark again, here are the results

jmh:run -i 3 -wi 3 -f1 -t4

Old code:
```
[info] Iteration   1: 379730,343 ops/s
[info] Iteration   2: 383230,774 ops/s
[info] Iteration   3: 381724,031 ops/s
[info] Result "sample.CookieDecodeBenchmark.benchmarkApp":
[info]   381561,716 ±(99.9%) 32033,267 ops/s [Average]
[info]   (min, avg, max) = (379730,343, 381561,716, 383230,774), stdev = 1755,851
[info]   CI (99.9%): [349528,449, 413594,983] (assumes normal distribution)
[info] # Run complete. Total time: 00:01:00
```

New code:
```
[info] Iteration   1: 653978,242 ops/s
[info] Iteration   2: 635232,010 ops/s
[info] Iteration   3: 633734,049 ops/s
[info] Result "sample.CookieDecodeBenchmark.benchmarkApp":
[info]   640981,434 ±(99.9%) 205797,553 ops/s [Average]
[info]   (min, avg, max) = (633734,049, 640981,434, 653978,242), stdev = 11280,458
[info]   CI (99.9%): [435183,881, 846778,986] (assumes normal distribution)
[info] # Run complete. Total time: 00:01:00

```